### ISSUE TITLE
Rendering/Layout

### ISSUE THREAD

**Issue by Place1** [Place1/react-native-gtk#25]
In this issue we should pool our ideas for how to design and implement the renderer and layout engine. Examples from other react renderers would be useful!

**Comment by clayrisser** [comment]
When I first started working on this, there were not many examples to pool from. I just checked and there seem to be quite a lot now. One that did exist when we first started on this project was tiny-react-renderer. It seemed pretty good.

https://github.com/iamdustan/tiny-react-renderer

**Comment by clayrisser** [comment]
Here's a tutorial . . .

https://github.com/nitin42/Making-a-custom-React-renderer

**Comment by Place1** [comment]
@codejamninja if you're looking for a challenge then this might interest you: https://github.com/kusti8/proton-native/pull/76

I've had a look at the implementation and it's a similar approach to what I was trying to do when experimenting. I think we can learn a lot from it.

**Comment by clayrisser** [comment]
Yeah, no prob. I've been playing around with react fiber and Yoga for a while. Trying to get a better grasp on it. I'm developing [react-gtk](https://github.com/codejamninja/react-gtk) to help me fully grasp react fiber.

I'm building react-gtk to match the GTK api as perfectly as possible. The react-native-gtk library will try and match the react native API as close as possible. I think there are benefits to both approaches.

Anyways, sorry it's a bit off topic. I'm kinda rambling about my current status on the project. I think there is a lot of knowledge I can contribute through experimentation. I'm def down for the challenge.

**Comment by clayrisser** [comment]
I'm also developing react-gtk in JavaScript instead of typescript, that way the type system doesn't get in the way of rapid prototyping. Take a look at the codebase, there are some architectural differences.
